test/integration/controller/system_controller_test.go

```diff
@@ -1209,6 +1219,115 @@ discovery:
 			return k8serrors.IsNotFound(err)
 		}, timeout, interval).Should(gomega.BeTrue())
 
+		resetMock(&styraClientMock.Mock)
+
+		key2 := types.NamespacedName{
+			Name:      "test2",
+			Namespace: "default",
+		}
+
+		toCreate2 := &styrav1beta1.System{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      key2.Name,
+				Namespace: key2.Namespace,
+			},
+			Spec: spec,
+		}
+		ginkgo.By("Connecting to a existing Styra System")
+
+		cfg := &styra.SystemConfig{
+			ID:   "system_id",
+			Name: key2.String(),
+		}
+
+		gomega.Expect(k8sClient.Create(ctx, toCreate2)).To(gomega.Succeed())
```

Comment:
We usually have this line after all the mock setup and right before the gomega.eventually function